logdetective/server/database/model.py

```diff
@@ -0,0 +1,85 @@
+import enum
+import datetime
+
+from typing import Optional
+from sqlalchemy import Column, Integer, Float, DateTime, String, Enum
+
+from logdetective.server.database.base import Base, transaction
+
+
+class AnalyzeRequestMetrics(Base):
+    """Store data related to received requests and given responses"""
+    __tablename__ = "analyze_request_metrics"
+
+    class EndpointType(enum.Enum):
+        """Different analyze endpoints"""
+        ANALYZE = "analyze_log"
+        ANALYZE_STAGED = "analyze_log_staged"
+        ANALYZE_STREAM = "analyze_log_stream"
```

Comment:
Nitpick: I dislike nesting classes and functions when possible. Can we please move this to top-level?

Reply:
Since I removed the definition of the database namespace I thought there is no reason to keep this. So I have moved the enum definition up.